src/hashing.rs

```diff
@@ -254,6 +254,49 @@ impl<D: Digest, M: DomainSeparation> DomainSeparatedHasher<D, M> {
     }
 }
 
+/// Implements Digest so that it can be used for other crates
```

Comment:
The `digest` crate already offers a default impl for 

```
impl<D: FixedOutput + Default + Update + HashMarker> Digest for D 
```

Is this intended to be a stopgap so `digest` is available while we look at impl the other trait bounds?

Reply:
I suppose I could implement `Update` and `Reset` individually instead, but I am not sure that covers all of them

Reply:
I tried implementing them individually, but because we need to push a prefix when we reset, the `Reset` needs `Update`, and `FixedOutput` needs `Reset` and it ended up being a lot more code than implementing the whole of `Digest`.